nomos-services/mempool/src/backend/mockpool.rs

```diff
@@ -128,3 +153,20 @@ where
             .collect()
     }
 }
+
+impl<BlockId, Item, Key> RecoverableMempool for MockPool<BlockId, Item, Key>
+where
+    Item: Clone + Send + Sync + 'static + Hash,
+    Key: Clone + Ord + Hash,
+    BlockId: Copy + Ord,
+{
+    type RecoveryState = Self;
+
+    fn recover(_settings: Self::Settings, state: Self::RecoveryState) -> Self {
+        state
+    }
+
+    fn save(&self) -> Self::RecoveryState {
+        self.clone()
+    }
```

Comment:
Yeah, since we clone here internally there should be no necessity for a bound in the outer layers (?)